File: problem_set2/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count(string text);

int main(void)
{
    string text = get_string("Text: ");

    int index = count(text);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n" , index);
    }
}
int count(string text)
{
    int n = strlen(text);
    float l = 0;
    float w = 1;
    float s = 0;
    for (int i = 0; i < n; i++)
    {
        if ((text [i] >= 65 && text [i] <= 90) || ( text [i] >= 97 && text [i] <= 122 ))
        {
            l++;
        }
        if (text [i] == 32)
        {
            w++;
        }
        if (text [i] == 63 || text [i] == 46 || text [i] == 33)
        {
            s++;
        }
    }
    float L = 100 * (l / w);
    float S = 100 * (s / w);
    float index = round(0.0588 * L - 0.296 * S - 15.8);
    return index;
}

File: CS50x/problem_set6/santimental-readability/readability.py
import string
from cs50 import get_string

text = get_string ("Text: ")
letters = 0
for i in text:
    if (i.isalpha()):
        letters +=1
words = len(text.split())
sentences = 0
for j in text:
    if (j == '.' or j == '!' or j == '?'):
        sentences +=1
w = words / 100
L = letters / w
S = sentences / w
index = round(0.0588 * L - 0.296 * S - 15.8)

if index < 1:
    print("Before Grade 1")
elif index >= 16:
    print ("Grade 16+")
else:
    print(f"Grade {index}")
